### PR TITLE
Rope permissions

### PR DESCRIPTION
**Pull request by iamn1ck**

### CODE REVIEW

**Comment by mosch** [comment]
🤷🏼‍♂️

File: src/rest/methods/user.ts [hunk 1]
```diff
@@ -50,13 +50,32 @@ export type UserResultList = Promise<{
...
-  assetContactPerson = 'asset-contact-person',
-  documentAdmin = 'document-admin',
-  externalTicketCollaborator = 'external-ticket-collaborator',
-  pinboardAdmin = 'community-article-admin',
+  articlesAgent = 'articles-agent',
+  bookableAssetAgent = 'bookable-asset-agent',
+  bookingAgent = 'booking-agent',
+  cockpitAdmin = 'cockpit-admin',
+  cockpitManager = 'cockpit-manager',
+  // communityArticleAdmin = 'community-article-admin',
```
Comment (adieuadieu) [inline src/rest/methods/user.ts]: Let's drop this and line 63 if we're not using them.

File: src/rest/methods/user.ts [hunk 2]
```diff
@@ -224,15 +243,53 @@ export async function userCreatePermission(
...
+    readonly objectType: EnumUserPermissionObjectType
+    readonly restrictions: ReadonlyArray<object>
+    readonly roles: ReadonlyArray<EnumUserPermissionRole>
+  },
+) => Promise<boolean>
+
+export async function userCreatePermissionBatch(
+  client: InterfaceAllthingsRestClient,
+  userId: string,
+  data: PartialUserPermission & {
```
Comment (adieuadieu) [inline src/rest/methods/user.ts]: On line 252 the function argument is called `permission` but here on line 262 it's called `data`. Let's make it the same for consistency/clarity sake—the convention so far appears to be to call it `data`.

File: src/rest/methods/user.ts [hunk 3]
```diff
@@ -224,15 +243,53 @@ export async function userCreatePermission(
...
+  const batch = {
+    batch: roles.map(role => ({
+      objectID: objectId,
+      objectType,
+      restrictions: [],
+      role,
+    })),
+  }
+
+  return !(await client.post(`/v1/users/${userId}/permissions`, batch))
```
Comment (dan-lee) [inline src/rest/methods/user.ts]: Why do you negate this here?

Comment (iamn1ck) [reply src/rest/methods/user.ts]: The batch permission endpoint returns an empty string as a result, so negating resolves to true

**Comment by iamn1ck** [comment]
Please refer to here for a list of the new roles 
https://github.com/allthings/php-prod/blob/dfe2ccaec3d4ae13221b6b6aca2c9a59092bdef6/symfony/src/Qipp/UserBundle/Model/RolePermissionV2Consts.php#L135

**Comment by adashst** [comment]
LGTM

**Pull request merged**